### PR TITLE
initial efforts to convert lnbitsbot-js to lnbitsbot-py

### PR DESCRIPTION
**Pull request by eriknewland**
I believe the core functionality is working, but still need to incorporate the buttons from telethon library. Also have a few methods left to solve involving the paylinks. 

bot.py:
api_id and api_hash are from the developer section on telegram. bot_token is from botfather. 

config.yml:
lnbits url, admin_key and in_key are from your lnbits wallet.

Notes:

Had a lot of difficulty with the decode_invoice function, believe the problem was a json parsing issue. Needed to make a slight change to pylnbits/utils.py (which im not sure is advised)

original
Screen Shot 2023-05-20 at 3 45 15 PM
my change
Screen Shot 2023-05-20 at 3 45 37 PM

### CODE REVIEW

**Comment by eriknewland** [comment]
I think there was another bug in user_wallet.py with check_invoice

Screen Shot 2023-05-21 at 5 53 06 PM
Screen Shot 2023-05-21 at 8 19 38 PM


Added a "main" /lightning route that displays buttons with the sub-routes/event handlers. You can still access these routes individually with their respective commands (i.e. clicking "create invoice" will trigger the same function as typing "/create_invoice"

Refactored general flow to be a question and answer sequence with bot rather than typing "/create_invoice <amount> <memo>"

Still have a couple methods to handle.

Need to handle exiting an event early (i.e. "quit/back/home" command that overrides current event)

Not sure about the presentation of some of the data that is being retrieving as I am still learning about lightning/lnbits API. 

Added BTC logo to QR code

File: bot.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+        # Ask for comment chars
+        await conv.send_message("<b>Finally, let's set the comment chars</b>", parse_mode="html")
+        comment_chars = int((await conv.get_response()).text)
+
+        # Create paylink
+        body = {
+            "description": description,
+            "amount": amount,
+            "max": max_amount,
```
Comment (bitkarrot) [inline bot.py:388]: max amt depends on server side limitations

File: bot.py [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+@client.on(events.NewMessage(pattern='/lightning'))
+async def lightning_menu(event):
+    buttons = [
+        [Button.inline("💰 Create Invoice", b"create_invoice")],
+        [Button.inline("💼 Get Balance", b"get_balance")],
+        [Button.inline("📝 Get Wallet Details", b"get_wallet_details")],
+        [Button.inline("🔍 Decode Invoice", b"decode_invoice")],
+        [Button.inline("💸 Pay Invoice", b"pay_invoice")],
+        [Button.inline("🔖 Check Invoice", b"check_invoice")],
+        [Button.inline("🔗 Create PayLink", b"create_paylink")],
```
Comment (bitkarrot) [inline bot.py:177]: suggest renaming PayLink to LNURLp.

**Review by bitkarrot — CHANGES_REQUESTED** [review]

**Comment by bitkarrot** [comment]
@eriknewland someone submitted a PR in the last day or two, this may help resolve the decode issue, https://github.com/lightningames/pylnbits/pull/17/files

File: bot.py (hunk 1 above)
Comment (eriknewland) [reply bot.py:394]: Noted but still a little unclear

**Comment by eriknewland** [comment]
Deployment to fly.io should work, though functionality is dubious. Couple of main issues.

1. Seems like there may be some kind of session issue with the pylnbits library I need to diagnose. Currently I get a run time error when attempting to create multiple invoices (or access multiple commands) back-to-back.

Screen Shot 2023-05-31 at 12 48 23 PM

2. In that same vein, noticed this issue when attempting to explore multiple commands

Screen Shot 2023-05-31 at 12 48 11 PM

What's strange is local deployment and the docker image seem to work fine without these issues. which makes me think that perhaps there's some issue with the fly.toml configuration. Will continue to investigate

**Comment by eriknewland** [comment]
needed to use `self.session` instead of `async with self.session`. weird this wasnt an issue locally.

**Pull request merged**